extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/job/DailyJobMoon.java

```diff
@@ -16,32 +18,98 @@
 import org.eclipse.smarthome.binding.astro.internal.model.Planet;
 
 /**
- * Schedules the events for the moon for the current day.
+ * Daily Scheduled Jobs For Moon Planet
  *
  * @author Gerhard Riegler - Initial contribution
+ * @author Amit Kumar Mondal - Implementation to be compliant with ESH Scheduler
  */
-public class DailyJobMoon extends AbstractDailyJob {
+public final class DailyJobMoon implements Job {
 
+    private final String thingUID;
+    private final AstroThingHandler handler;
+
+    /**
+     * Constructor
+     *
+     * @param thingUID
+     *            the Thing UID
+     * @param handler
+     *            the {@link AstroThingHandler} instance
+     * @throws NullPointerException
+     *             if {@code thingUID} or {@code handler} is {@code null}
+     */
+    public DailyJobMoon(String thingUID, AstroThingHandler handler) {
+        checkNotNull(thingUID, "Thing UID cannot be null");
```

Comment:
Please change it to a normal `null` check and throw a `IllegalArgumentException`. This is more meaningful than throwing an NPE here.